sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -150,7 +179,359 @@ class KmacTest : public testing::Test, public mock_mmio::MmioTest {
+          config_.en_unsupported_modestrength}});
+  }
+
+  void SetExpectedKey(const dif_kmac_key_t &key) {
+    std::map<dif_kmac_key_length_t, uint32_t> keySize = {
+        {kDifKmacKeyLen128, KMAC_KEY_LEN_LEN_VALUE_KEY128},
+        {kDifKmacKeyLen192, KMAC_KEY_LEN_LEN_VALUE_KEY192},
+        {kDifKmacKeyLen256, KMAC_KEY_LEN_LEN_VALUE_KEY256},
+        {kDifKmacKeyLen384, KMAC_KEY_LEN_LEN_VALUE_KEY384},
+        {kDifKmacKeyLen512, KMAC_KEY_LEN_LEN_VALUE_KEY512},
+    };
+
+    EXPECT_WRITE32(KMAC_KEY_LEN_REG_OFFSET, keySize[key.length]);
+    for (uint32_t i = 0; i < ARRAYSIZE(key.share0); ++i) {
+      ptrdiff_t offset = KMAC_KEY_SHARE0_0_REG_OFFSET + (i * sizeof(uint32_t));
+      EXPECT_WRITE32(offset, key.share0[i]);
+      offset = KMAC_KEY_SHARE1_0_REG_OFFSET + (i * sizeof(uint32_t));
+      EXPECT_WRITE32(offset, key.share1[i]);
+    }
+  }
+
+  void SetExpectedPrefix(const uint32_t *prefix_regs, uint32_t size) {
+    for (uint32_t i = 0; i < size; ++i) {
+      ptrdiff_t offset = KMAC_PREFIX_0_REG_OFFSET + (i * sizeof(uint32_t));
+      EXPECT_WRITE32(offset, prefix_regs[i]);
+    }
+  }
+};
+
+class Kmac256Test : public KmacTest {
+ protected:
+  dif_kmac_key_t key_ = {
+                .share0 = {0x43424140, 0x47464544, 0x4B4A4948, 0x4F4E4D4C,
+                           0x53525150, 0x57565554, 0x5B5A5958, 0x5F5E5D5C},
+                .share1 = {0},
+                .length = kDifKmacKeyLen256,
+            };
+  dif_kmac_mode_kmac_t mode_ = kDifKmacModeKmacLen256;
+  dif_kmac_customization_string_t custom_string_;
+
+  Kmac256Test() {
+    const std::string kString = "My Tagged Application";
+    EXPECT_DIF_OK(dif_kmac_customization_string_init(
+        kString.c_str(), kString.size(), &custom_string_));
+    config_.en = true;
+    config_.mode = KMAC_CFG_SHADOWED_MODE_VALUE_CSHAKE;
+  }
+  void SetExpectedPrefix(const dif_kmac_customization_string_t &s) {
+    // Initialize prefix registers with function name ("KMAC") and empty
+    // customization string. The empty customization string will be overwritten
+    // if a non-empty string is provided.
+    uint32_t prefix_regs[11] = {
+        0x4D4B2001,  //  1  32  'K' 'M'
+        0x00014341,  // 'A' 'C'  1   0
+    };
+
+    // Encoded customization string (s) must be at least 3 bytes long if it is
+    // not the empty string.
+    if (s.length >= 3) {
+      // First two bytes overwrite the pre-encoded empty customization string.
+      prefix_regs[1] &= 0xFFFF;
+      prefix_regs[1] |= (uint32_t)((uint8_t)s.buffer[0]) << 16;
+      prefix_regs[1] |= (uint32_t)((uint8_t)s.buffer[1]) << 24;
+      memcpy(&prefix_regs[2], &s.buffer[2], s.length - 2);
+    }
+
+    KmacTest::SetExpectedPrefix(prefix_regs, ARRAYSIZE(prefix_regs));
+  }
+};
+
+TEST_F(Kmac256Test, StartSuccess) {
+  EXPECT_READ32(KMAC_STATUS_REG_OFFSET, {{KMAC_STATUS_SHA3_IDLE_BIT, true}});
+  SetExpectedKey(key_);
+  EXPECT_READ32(KMAC_CFG_SHADOWED_REG_OFFSET, 0);
+  setExpectedConfig();
```

Comment:
SetExpectedConfig; note the leading S.

I've found that we generally like to call functions that set EXPECTs something like `ExpectFoo()`.